instrumentation/docker/src/main/scala/net/cakesolutions/testkit/logging/docker/DockerLogSource.scala

```diff
@@ -0,0 +1,96 @@
+// Copyright 2018 Cake Solutions Limited
+// Copyright 2017 Carl Pulley
+
+package net.cakesolutions.testkit.logging.docker
+
+import scala.concurrent.{blocking, Promise}
+import scala.sys.process.{Process, ProcessLogger}
+import scala.util.control.NonFatal
+
+import com.typesafe.scalalogging.Logger
+import io.circe.Json
+import monix.execution.Scheduler
+import monix.reactive.observers.Subscriber
+
+import net.cakesolutions.testkit.config.Configuration.Logging
+import net.cakesolutions.testkit.logging.{LogEvent, LoggingSource}
+import net.cakesolutions.testkit.logging.docker.DockerLogSource.ProcessTerminated
+import net.cakesolutions.testkit.logging.docker.formats.LogEventFormat
+
+trait BaseDockerLogSource extends LoggingSource[Json] {
```

Comment:
Hmm, I considered a split like this, but constructed as a generic `ProcessObservable`, with a `DockerLogSource` based on it.